--- a/config.h	2020-10-08 16:56:50.000000000 +0900
+++ b/config.h	2020-10-08 16:57:01.000000000 +0900
@@ -733,7 +733,7 @@
 // Limit the <arm_acle.h> include.
 #if !defined(CRYPTOPP_ARM_ACLE_AVAILABLE)
 # if defined(__aarch32__) || defined(__aarch64__) || (__ARM_ARCH >= 8) || defined(__ARM_ACLE)
-#  if !defined(__ANDROID__) && !defined(ANDROID) && !defined(__APPLE__)
+#  if !defined(__ANDROID__) && !defined(ANDROID)
 #   define CRYPTOPP_ARM_ACLE_AVAILABLE 1
 #  endif
 # endif
