--- a/cctools/include/stuff/diagnostics.h	2022-01-15 17:21:43.000000000 +0900
+++ b/cctools/include/stuff/diagnostics.h	2022-01-15 17:22:17.000000000 +0900
@@ -60,13 +60,6 @@
  */
 void diagnostics_write(void);
 
-#if defined(__APPLE__ ) && defined(__has_builtin)
-#  if __has_builtin(__builtin_available)
-#    define HAVE_OPENMEMSTREAM_RUNTIME __builtin_available(macOS 10.13, *)
-#  endif
-#endif
-#ifndef HAVE_OPENMEMSTREAM_RUNTIME
-#  define HAVE_OPENMEMSTREAM_RUNTIME 1
-#endif
+#define HAVE_OPENMEMSTREAM_RUNTIME 0
 
 #endif /* diagnostics_h */
