--- a/src/auto/configure
+++ b/src/auto/configure
@@ -4587,128 +4587,6 @@
 $as_echo "yes, Darwin support excluded" >&6; }
   fi
 
-  { $as_echo "$as_me:${as_lineno-$LINENO}: checking --with-developer-dir argument" >&5
-$as_echo_n "checking --with-developer-dir argument... " >&6; }
-
-# Check whether --with-developer-dir was given.
-if test "${with_developer_dir+set}" = set; then :
-  withval=$with_developer_dir; DEVELOPER_DIR="$withval"; { $as_echo "$as_me:${as_lineno-$LINENO}: result: $DEVELOPER_DIR" >&5
-$as_echo "$DEVELOPER_DIR" >&6; }
-else
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: not present" >&5
-$as_echo "not present" >&6; }
-fi
-
-
-  if test "x$DEVELOPER_DIR" = "x"; then
-    # Extract the first word of "xcode-select", so it can be a program name with args.
-set dummy xcode-select; ac_word=$2
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
-$as_echo_n "checking for $ac_word... " >&6; }
-if ${ac_cv_path_XCODE_SELECT+:} false; then :
-  $as_echo_n "(cached) " >&6
-else
-  case $XCODE_SELECT in
-  [\\/]* | ?:[\\/]*)
-  ac_cv_path_XCODE_SELECT="$XCODE_SELECT" # Let the user override the test with a path.
-  ;;
-  *)
-  as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
-for as_dir in $PATH
-do
-  IFS=$as_save_IFS
-  test -z "$as_dir" && as_dir=.
-    for ac_exec_ext in '' $ac_executable_extensions; do
-  if as_fn_executable_p "$as_dir/$ac_word$ac_exec_ext"; then
-    ac_cv_path_XCODE_SELECT="$as_dir/$ac_word$ac_exec_ext"
-    $as_echo "$as_me:${as_lineno-$LINENO}: found $as_dir/$ac_word$ac_exec_ext" >&5
-    break 2
-  fi
-done
-  done
-IFS=$as_save_IFS
-
-  ;;
-esac
-fi
-XCODE_SELECT=$ac_cv_path_XCODE_SELECT
-if test -n "$XCODE_SELECT"; then
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: $XCODE_SELECT" >&5
-$as_echo "$XCODE_SELECT" >&6; }
-else
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-fi
-
-
-    if test "x$XCODE_SELECT" != "x"; then
-      { $as_echo "$as_me:${as_lineno-$LINENO}: checking for developer dir using xcode-select" >&5
-$as_echo_n "checking for developer dir using xcode-select... " >&6; }
-      DEVELOPER_DIR=`$XCODE_SELECT -print-path`
-      { $as_echo "$as_me:${as_lineno-$LINENO}: result: $DEVELOPER_DIR" >&5
-$as_echo "$DEVELOPER_DIR" >&6; }
-    else
-      DEVELOPER_DIR=/Developer
-    fi
-  fi
-
-  { $as_echo "$as_me:${as_lineno-$LINENO}: checking --with-macsdk argument" >&5
-$as_echo_n "checking --with-macsdk argument... " >&6; }
-
-# Check whether --with-macsdk was given.
-if test "${with_macsdk+set}" = set; then :
-  withval=$with_macsdk;  MACSDK="$withval"; { $as_echo "$as_me:${as_lineno-$LINENO}: result: $MACSDK" >&5
-$as_echo "$MACSDK" >&6; }
-else
-   MACSDK=""; { $as_echo "$as_me:${as_lineno-$LINENO}: result: using default" >&5
-$as_echo "using default" >&6; }
-fi
-
-
-  if test -n "$MACSDK"; then
-    { $as_echo "$as_me:${as_lineno-$LINENO}: checking if SDK is supported" >&5
-$as_echo_n "checking if SDK is supported... " >&6; }
-    save_cflags="$CFLAGS"
-    save_ldflags="$LDFLAGS"
-    sdkflags="$DEVELOPER_DIR/SDKs/MacOSX$MACSDK"
-    if test "x$MACSDK" = "x10.4"; then
-      sdkflags="$sdkflags""u"
-    fi
-    sdkflags="$sdkflags.sdk -mmacosx-version-min=$MACSDK"
-    CFLAGS="$CFLAGS -isysroot $sdkflags"
-    LDFLAGS="$LDFLAGS -Wl,-syslibroot,$sdkflags"
-    cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-int
-main ()
-{
-
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
-   { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
-$as_echo "yes" >&6; }
-        XCODEFLAGS="$XCODEFLAGS -sdk macosx$MACSDK MACOSX_DEPLOYMENT_TARGET=$MACSDK"
-else
-   { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-                        MACSDK=""
-        CFLAGS="$save_cflags"
-        LDFLAGS="$save_ldflags"
-fi
-rm -f core conftest.err conftest.$ac_objext \
-    conftest$ac_exeext conftest.$ac_ext
-  else
-    if test -z "$MACOSX_DEPLOYMENT_TARGET"; then
-      macosx_deployment_target=`/usr/bin/sw_vers -productVersion|/usr/bin/sed -e 's/^\([0-9]*\.[0-9]*\).*/\1/'`
-      XCODEFLAGS="$XCODEFLAGS MACOSX_DEPLOYMENT_TARGET=$macosx_deployment_target"
-    fi
-  fi
-
-
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking --with-macarchs argument" >&5
 $as_echo_n "checking --with-macarchs argument... " >&6; }
 
@@ -5828,10 +5706,7 @@
 
     for path in "${vi_cv_path_mzscheme_pfx}/lib" "${SCHEME_LIB}"; do
       if test "X$path" != "X"; then
-	if test "x$MACOS_X" = "xyes"; then
-	  MZSCHEME_LIBS="-framework Racket"
-	  MZSCHEME_CFLAGS="-DMZ_PRECISE_GC"
-	elif test -f "${path}/libmzscheme3m.a"; then
+	if test -f "${path}/libmzscheme3m.a"; then
 	  MZSCHEME_LIBS="${path}/libmzscheme3m.a"
 	  MZSCHEME_CFLAGS="-DMZ_PRECISE_GC"
 	elif test -f "${path}/libracket3m.a"; then
@@ -6216,23 +6091,6 @@
   fi
 
   if test "x$MACOS_X" = "xyes"; then
-        dir=/System/Library/Perl
-    darwindir=$dir/darwin
-    if test -d $darwindir; then
-      PERL=/usr/bin/perl
-    else
-            dir=/System/Library/Perl/5.8.1
-      darwindir=$dir/darwin-thread-multi-2level
-      if test -d $darwindir; then
-	PERL=/usr/bin/perl
-      fi
-    fi
-    if test -n "$PERL"; then
-      PERL_DIR="$dir"
-      PERL_CFLAGS="-DFEAT_PERL -I$darwindir/CORE"
-      PERL_OBJ="objects/if_perl.o objects/if_perlsfio.o $darwindir/auto/DynaLoader/DynaLoader.a"
-      PERL_LIBS="-L$darwindir/CORE -lperl"
-    fi
                 PERL_LIBS=`echo "$PERL_LIBS" | sed -e 's/-arch\ ppc//' -e 's/-arch\ i386//' -e 's/-arch\ x86_64//'`
     PERL_CFLAGS=`echo "$PERL_CFLAGS" | sed -e 's/-arch\ ppc//' -e 's/-arch\ i386//' -e 's/-arch\ x86_64//'`
   fi
@@ -6455,13 +6313,6 @@
 eof
 	    	    eval "`cd ${PYTHON_CONFDIR} && make -f "${tmp_mkf}" __ | sed '/ directory /d'`"
 	    rm -f -- "${tmp_mkf}"
-	    if test "x$MACOS_X" = "xyes" && test -n "${python_PYTHONFRAMEWORK}" && ${vi_cv_path_python} -c \
-		"import sys; sys.exit(${vi_cv_var_python_version} < 2.3)"; then
-	      vi_cv_path_python_plibs="-framework Python"
-	      if test "x${vi_cv_path_python}" != "x/usr/bin/python" && test -n "${python_PYTHONFRAMEWORKPREFIX}"; then
-		  vi_cv_path_python_plibs="-F${python_PYTHONFRAMEWORKPREFIX} -framework Python"
-	      fi
-	    else
 	      vi_cv_path_python_plibs="-L${PYTHON_CONFDIR} -lpython${vi_cv_var_python_version}"
 	      	      	      	      if test -n "${python_LINKFORSHARED}" && test -n "${python_PYTHONFRAMEWORKPREFIX}"; then
 	        python_link_symbol=`echo ${python_LINKFORSHARED} | sed 's/\([^ \t][^ \t]*[ \t][ \t]*[^ \t][^ \t]*\)[ \t].*/\1/'`
@@ -6476,7 +6327,6 @@
 	      fi
 	      vi_cv_path_python_plibs="${vi_cv_path_python_plibs} ${python_BASEMODLIBS} ${python_LIBS} ${python_SYSLIBS} ${python_LINKFORSHARED}"
 	      	      vi_cv_path_python_plibs=`echo $vi_cv_path_python_plibs | sed s/-ltermcap//`
-	    fi
 
 fi
 
@@ -6555,13 +6405,6 @@
 $as_echo "no" >&6; }
 	fi
 
-                                if test -n "$MACSDK"; then
-          PYTHON_CFLAGS=
-          PYTHON_LIBS=-framework Python
-          PYTHON_CONFDIR=
-          PYTHON_GETPATH_CFLAGS=
-        fi
-
 			{ $as_echo "$as_me:${as_lineno-$LINENO}: checking if compile and link flags for Python are sane" >&5
 $as_echo_n "checking if compile and link flags for Python are sane... " >&6; }
 	cflags_save=$CFLAGS
@@ -7455,11 +7298,7 @@
 
       { $as_echo "$as_me:${as_lineno-$LINENO}: checking for location of Tcl include" >&5
 $as_echo_n "checking for location of Tcl include... " >&6; }
-      if test "x$MACOS_X" != "xyes"; then
 	tclinc="$tclloc/include $tclloc/include/tcl $tclloc/include/tcl$tclver /usr/local/include /usr/local/include/tcl$tclver /usr/include /usr/include/tcl$tclver"
-      else
-		tclinc="/System/Library/Frameworks/Tcl.framework/Headers"
-      fi
       TCL_INC=
       for try in $tclinc; do
 	if test -f "$try/tcl.h"; then
@@ -7477,12 +7316,8 @@
       if test -z "$SKIP_TCL"; then
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for location of tclConfig.sh script" >&5
 $as_echo_n "checking for location of tclConfig.sh script... " >&6; }
-	if test "x$MACOS_X" != "xyes"; then
 	  tclcnf=`echo $tclinc | sed s/include/lib/g`
 	  tclcnf="$tclcnf `echo $tclinc | sed s/include/lib64/g`"
-	else
-	  	  tclcnf="/System/Library/Frameworks/Tcl.framework"
-	fi
 	for try in $tclcnf; do
 	  if test -f "$try/tclConfig.sh"; then
 	    { $as_echo "$as_me:${as_lineno-$LINENO}: result: $try/tclConfig.sh" >&5
@@ -7672,10 +7507,6 @@
 	if test -f "$rubylibdir/$librubya"; then
 	  librubyarg="$librubyarg"
 	  RUBY_LIBS="$RUBY_LIBS -L$rubylibdir"
-        elif test "$vi_cv_path_ruby" = "/usr/bin/ruby" -a -d "/System/Library/Frameworks/Ruby.framework"; then
-                        RUBY_LIBS="-framework Ruby"
-                        RUBY_CFLAGS="$RUBY_CFLAGS -DRUBY_VERSION=$rubyversion"
-            librubyarg=
 	fi
 
 	if test "X$librubyarg" != "X"; then
--- a/src/vimtutor
+++ b/src/vimtutor
@@ -12,7 +12,7 @@
 # Vim could be called "vim" or "vi".  Also check for "vimN", for people who
 # have Vim installed with its version number.
 # We anticipate up to a future Vim 8.1 version :-).
-seq="vim vim81 vim80 vim8 vim74 vim73 vim72 vim71 vim70 vim7 vim6 vi"
+seq="mvim vim vim81 vim80 vim8 vim74 vim73 vim72 vim71 vim70 vim7 vim6 vi"
 if test "$1" = "-g"; then 
   # Try to use the GUI version of Vim if possible, it will fall back
   # on Vim if Gvim is not installed.
